test/parsing/ferguson/this-intent.chpl

```diff
@@ -0,0 +1,22 @@
+record R {
+  var x:int;
+  proc f() {
+    writeln(x);
+  }
+  proc ref g() {
+    writeln(x);
+  }
+  proc const h() {
+    writeln(x);
+  }
+  proc const ref i() {
+    writeln(x);
+  }
```

Comment:
I take that back - it does work with 'const ref' but not 'const'. I'll file a test and a .future in this PR about it and make a note to myself to try to fix the 'const' case in the umbrella PR.